src/app/modules/time-entries/pages/time-entries.component.html

```diff
@@ -54,8 +54,14 @@
             <th class="col"></th>
           </tr>
         </thead>
+        <tr *ngIf="checkIfDataSourceIsLoadingorEmpty(dataSource)">
+          <td class="text-center" colspan="7">{{NO_DATA_MESSAGE}}</td>
+        </tr>
         <app-loading-bar *ngIf="dataSource.isLoading"></app-loading-bar>
         <tbody *ngIf="!dataSource.isLoading">
+          <tr *ngIf="!checkIfDataSourceIsLoadingorEmpty(dataSource)">
```

Comment:
I'm not sure if this implementation is correct, for example in line `61`, you do this:
`<tbody *ngIf="!dataSource.isLoading">`
Here you check if dataSOurce is loading, and this extract of code only appear if dataSource loading has been completed (i.e. `false`), however in line `62`, that method `checkIfDataSourceIsLoadingorEmpty(dataSource)`, verifies again if dataSource is loading, I think that implementation is incorrect, because it does not make sense to perform the verification again if you know that dataSource.isLoading variable isi `false`

Reply:
Thanks for your review. Line 61 was not modified, and line 62 will have another approach.